Makefile

```diff
@@ -1,16 +1,26 @@
 SHELL = /bin/bash
 LANG = en_US.utf-8
-PYTHON = $(shell which python3 || which python)
+PYTHON = $(shell which python3.10 || which python3 || which python)
 DOCKER = $(shell which docker)
-DOCKER_COMPOSE = $(shell which docker || echo "$(DOCKER) compose")
+DOCKER_COMPOSE = $(shell $(DOCKER) compose > /dev/null && echo "$(DOCKER) compose" || which docker-compose)
 export LANG
 
 # Initializes submodules and copies environment file sample to env file.
 .PHONY:init
 init:.env
-	poetry install
-	poetry run pre-commit install
-	git submodule update --init;
+	@if ($(PYTHON) -m poetry -q > /dev/null); then \
+		$(PYTHON) -m poetry install; \
```

Comment:
This might be too dependent on a specific path setup. I have a few installations of Python and I run `poetry` by just using `poetry install` and not `python3 -m poetry install`. The latter doesn't work on my computer because the default python installation doesn't allow packages to be installed.